README.md

```diff
@@ -387,6 +391,62 @@ The [./examples/help/](./examples/help/) folder contains a working example of
 command that shows a generic help/usage message and also specific component
 help messages when used with `-help <ComponentName>`
 
+
+## Provided Baker components
+
+### Inputs
+
+#### KCL
+
+`input.KCL` fetches records from AWS [Kinesis](https://aws.amazon.com/kinesis/)
+using [vmware-go-kcl](https://github.com/vmware/vmware-go-kcl), an
+implementation of the KCL (Kinesis Client Library).
+KCL provides a way to to process a single Kinesis stream from multiple Baker
+instances, each instances reading a specific set of shards.
+
+The KCL takes care of balancing the shards between workers. At the time of
+writing, vmware-go-kcl doesn't implement shard stealing yet, so it's advised to
+set MaxShards to a reasonable value. Since the number of shards doesn't change
+often, dividing the number of total shards by the expected number of baker
+instances and rounding up to the next integer has given us good results.
+
+The association between shards and baker instances (or workers) are called
+leases. Lease synchronization is taken care of by KCL; to do so it requires
+access to a DynamoDB table, which named depends on the configured AppName.
+
+Leases are updated at regular interval defined by ShardSync.
+
+The dynamodb table also serves the purpose of checkpointing, that is keeping
+track of the per-shard advancement by writing the ID last read record
+(checkpoint).
+
+You can choose the initial position to read from when the application starts
+(that is when the dynamodb table doesn't exist) by setting InitialPosition
+either to LATEST or TRIM_HORIZON.
```

Comment:
I'd add a suggestion for the value to use (rephrase as you like the most):


Note that those values are used only when a checkpoint doesn't exist in DynamoDB for the shard. If the checkpoint exists, its value is always used first.

##### How to choose the correct InitialPosition value 

When chosing the InitialPosition value, have in mind that `TRIM_HORIZON` can be useful when a resharding happens. In that case, in fact, before baker starts reading the new shards, some records could have already been published there. In this case using `LATEST` leads to have those records ignored. On the other hand, using `TRIM_HORIZON` means that, the first time that baker starts reading an existing stream, all the records hold in the shards must be read (data retention period depends on the stream configuration and is variable between 24 hours to 7 days). If reading all the existing records is a problem, then a possible strategy is to start baker the first time with `LATEST` (to create the first checkpoint) and then switch to `TRIM_HORIZON`